crates/wash/src/lib/start/nats.rs

```diff
@@ -378,7 +398,14 @@ fn nats_url(os: &str, arch: &str, version: &str) -> String {
         "x86_64" => "amd64",
         _ => arch,
     };
-    format!("{NATS_GITHUB_RELEASE_URL}/{version}/nats-server-{version}-{os}-{arch}.tar.gz")
+    if let Some(mirror) = mirror {
```

Comment:
Super duper nit: I think with all this munging of paths and URLs and stuff, it might be a bit safer to use something like [`url`](https://docs.rs/url/latest/url/). But honestly if it is just too much of a hassle, don't worry about it